scripts/check-docs-quality causes vale to segfault

<!--- Provide a general summary of the issue in the Title above -->

`yarn run lint:docs` is causing a vale segfault on macOS

## Expected Behavior

<!--- Tell us what should happen -->

```
backstage$ yarn run lint:docs
yarn run v1.22.1
$ node ./scripts/check-docs-quality
✔ 0 errors, 0 warnings and 0 suggestions in 336 files.
✨  Done in 1.93s.
backstage$ 
```
## Current Behavior

<!--- Tell us what happens instead of the expected behavior -->

```
yarn run lint:docs
yarn run v1.22.1
$ node ./scripts/check-docs-quality
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x16eb9cd]

goroutine 19226 [running]:
github.com/errata-ai/vale/lint.(*Linter).lintFiles.func1.1(0x7ffeefbfe2e5, 0x0, 0x0, 0x0, 0x1a48200, 0xc001f30f60, 0xc00012b320, 0x1824700)
        github.com/errata-ai/vale/lint/lint.go:126 +0x4d
path/filepath.Walk(0x7ffeefbfe2e5, 0x0, 0xc000116f78, 0x208f9c0, 0x208f9c0)
        path/filepath/path.go:404 +0x6b
github.com/errata-ai/vale/lint.(*Linter).lintFiles.func1(0x7ffeefbfe2e5, 0x0, 0xc000ae6300, 0xc00012b2c0, 0xc00012a180, 0xc001c69500)
        github.com/errata-ai/vale/lint/lint.go:124 +0x114
created by github.com/errata-ai/vale/lint.(*Linter).lintFiles
        github.com/errata-ai/vale/lint/lint.go:121 +0xb6
Language linter (vale) generated errors. Please check the errors and review any markdown files that you changed.
  Possibly update .github/styles/vocab.txt to add new valid words.

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

## Possible Solution

<!--- Not obligatory, but suggest a fix/reason for the bug, -->
<!--- or ideas as to the implementation of the addition or change -->

The segfault is caused by an empty filename being passed to vale due to an array resulting from the split of `git ls-files` ending with an empty element at https://github.com/spotify/backstage/blob/master/scripts/check-docs-quality.js#L43-L48

(PR coming shortly.)

## Steps to Reproduce

<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code or configuration to reproduce, if relevant -->

1. `yarn run lint:docs`

## Context

<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context (e.g. links to configuration settings, -->
<!--- stack trace or log data) helps us come up with a solution that is most useful in the real world -->

I was trying to submit a documentation update PR #3187.

## Your Environment

<!--- Include as many relevant details about the environment you experienced the bug in -->

- NodeJS Version (v12): v12.18.3
- Operating System and Version (e.g. Ubuntu 14.04): macOS Catalina 10.15.7 
- Browser Information: none

--- a/scripts/check-docs-quality.js
+++ b/scripts/check-docs-quality.js
@@ -44,7 +44,10 @@ const getFilesToLint = () => {
     stdio: ['ignore', 'pipe', 'inherit'],
   })
     .toString()
-    .split('\n');
+    .split('\n')
+    .filter(function (el) {
+      return el != '';
+    });
 };
 
 // Proceed with the script only if Vale linter is installed. Limit the friction and surprises caused by the script.

